control-plane/plugin/src/resources/node.rs

```diff
@@ -378,77 +413,93 @@ impl Drain for Node {
-                                        }
-                                        break;
+                                }
+                                break;
+                            }
+                            CordonDrainState::drainingstate(_) => {}
+                            CordonDrainState::drainedstate(_) => {
+                                match output {
+                                    OutputFormat::None => {
+                                        println!("Node {} successfully drained", id);
                                     }
-                                },
-                                None => {
-                                    match output {
-                                        OutputFormat::None => {
-                                            println!("Node {} drain has been cancelled", id);
-                                        }
-                                        _ => {
-                                            // json or yaml
-                                            print_table(output, node.into_body());
-                                        }
+                                    _ => {
+                                        // json or yaml
+                                        print_table(output, node.into_body());
                                     }
-                                    break;
+                                }
+                                break;
+                            }
+                        },
+                        None => {
+                            match output {
+                                OutputFormat::None => {
+                                    println!("Node {} drain has been cancelled", id);
+                                }
+                                _ => {
+                                    // json or yaml
+                                    print_table(output, node.into_body());
                                 }
                             }
-                        }
-                        Err(e) => {
-                            println!("Failed to get node {}. Error {}", id, e);
                             break;
                         }
                     }
-                    if timeout_instant.is_some() && time::Instant::now() > timeout_instant.unwrap()
-                    {
-                        println!("Node {} drain command timed out", id);
-                        break;
-                    }
-                    let sleep = Duration::from_secs(1);
-                    tokio::time::sleep(sleep).await;
+                }
+                Err(e) => {
+                    println!("Failed to get node {}. Error {}", id, e);
+                    break;
                 }
             }
-            Err(e) => {
-                println!("Failed to get node {}. Error {}", id, e)
+            if timeout_instant.is_some() && time::Instant::now() > timeout_instant.unwrap() {
```

Comment:
nit: btw you could also "bundle" the code above into a future, and then call it with tokio timeout, which will automatically cancel the future and return when the timeout is reached.